### PR TITLE
Add ID type as per the GraphQL spec

### PR DESCRIPTION
**Pull request by pushrax**
Ref: http://facebook.github.io/graphql/#sec-ID

### CODE REVIEW

File: lib/graph_ql/types/id_type.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+GraphQL::ID_TYPE = GraphQL::ScalarType.new do |t|
+  t.name "ID"
+  def t.coerce(value)
+    value.to_s
```
Comment (pushrax) [inline lib/graph_ql/types/id_type.rb]: > While it is often numeric, it should always serialize as a String.

File: spec/support/dummy_app.rb [hunk 2]
```diff
@@ -93,7 +93,7 @@ def description
   end
 
   def resolve(target, arguments, context)
-    @data[arguments["id"]]
+    @data[arguments["id"].to_i]
```
Comment (pushrax) [inline spec/support/dummy_app.rb]: > GraphQL servers should coerce as appropriate given the ID formats they expect

File: spec/graph_ql/query_spec.rb [hunk 3]
```diff
@@ -50,6 +50,26 @@
       assert_equal(GraphQL::Nodes::FragmentDefinition, query.fragments['cheeseFields'].class)
     end
 
+    describe 'scalar coercion' do
```
Comment (rmosolgo) [inline spec/graph_ql/query_spec.rb]: Could you create  `spec/graph_ql/types/id_type_spec.rb` to test this behavior? Fine with me to just copy-paste the other `let`s etc. 

Even though `Query` is one of the hotspots, I'm hoping to keep the spec pretty slim and test other behaviors in specific places!

**Comment by rmosolgo** [comment]
:+1: looks awesome, thanks for getting us up to spec!

File: spec/graph_ql/query_spec.rb (hunk 3 above)
Comment (pushrax) [inline spec/graph_ql/query_spec.rb]: Moved to `scalar_type_spec.rb`, which I think makes sense (the scalars are of type `ScalarType`) and will remain small.

**Comment by rmosolgo** [comment]
:confetti_ball:  :tada: :beers:

**Pull request merged**